.github/workflows/auto_deploy_staging.yaml

```diff
@@ -0,0 +1,32 @@
+name: Apply changes
+
+# only trigger on pull request closed events
+on:
+  pull_request:
+    types: [ closed ]
+    branches:
+      - staging
```

Comment:
This does work as expected - the action only gets triggered when merging into staging.